specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,509 @@
+    notification.Tag(friendId);
+
+    // Developer sets the Expiry property on reboot to true to delete the notification on reboot
+    notification.ExpiresOnReboot(true);
+    winrt::AppNotificationManager::Default().Show(notification);
+}
+```
+
+## Processing a user generated Action on the Notification
+
+The app will need to process the invocation in response to a User interacting with the Notification
+in Action Centre. There are 2 common scenarios here after the invoke takes place:
+
+1. The developer can choose to have the app launch in in a specific UI context OR
+2. The developer can choose to have the app evaluate an action specific behavior (like a button
+   press in the payload body) without rendering any UI.
+
+The example below goes through a scenario where the developer evaluates the Activation Arguments to
+determine whether a foreground window needs to be rendered or whether some background business logic
+needs to be run without any special UI rendering.
+
+Note: activationType="background" in the Notification payload body is ignored for Desktop apps. It
+is upto the developer to process the Activation Arguments and decide whether to display UI or not.
+
+```cpp
+//<toast launch="action=openThread&amp;threadId=92187">
+//  <visual>
+//      <binding template="ToastGeneric">
+//          <text hint-maxLines="1">Jill Bender</text>
+//          <text>Check out where we camped last weekend! Incredible!!</text>
+//          <image placement="hero" src="https://unsplash.it/360/180?image=1043" />
+//      </binding>
+//  </visual>
+//  <actions>
+//      <input id="textBox" type="text" placeHolderContent="reply" / >
+//      <action
+//          content="Send"
+//          imageUri="Assets/Icons/send.png"
+//          hint-inputId="textBox"
+//          activationType="background"
+//          arguments="action=reply&amp;threadId=92187" />
+//  </actions>
+//</toast>
+
+void ProcessNotificationArgs(const winrt::AppNotificationActivatedEventArgs& notificationActivatedEventArgs)
+{
+    // If the user clicks on a toast, the code will need to launch the chat thread window
+    if (std::wstring(notificationActivatedEventArgs.ActivationArgs().c_str()).find(L"openThread") != std::wstring::npos)
+    {
+        GenerateChatThreadWindow();
+    }
+    else // If the user responds to a message by clicking a button in the notification, we will need to reply back to the other user
+    if (std::wstring(notificationActivatedEventArgs.ActivationArgs().c_str()).find(L"reply") != std::wstring::npos)
+    {
+        auto input = notificationActivatedEventArgs.UserInput();
+        auto replyBoxText = input.Lookup(L"ReplyBox");
+
+        // Process the reply text
+        SendReplyToUser(replyBoxText);
+    }
+}
+```
+
+## Processing Active Notifications
+
+We define Active Notifications as payloads that have already been displayed by the app and show up
```

Comment:
"We define..." is unnatural here. and describing a toast as a "payload" is indirect.